Remove RepaintOutsideLayoutEnabled setting

anttijk opened the pull request:
#### ba6ec09c50921ea86aa60a084f90b4022bc8d721
<pre>
Remove RepaintOutsideLayoutEnabled setting
<a href="https://bugs.webkit.org/show_bug.cgi?id=258570">https://bugs.webkit.org/show_bug.cgi?id=258570</a>
rdar://111390032

Reviewed by Alan Baradlay.

Remove a long-dead experiment.

* Source/WebCore/page/Settings.yaml:
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::checkForRepaintDuringLayout const):

Canonical link: <a href="https://commits.webkit.org/265554@main">https://commits.webkit.org/265554@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7500d61cff4a36f78bf14ffaf4ac8504fbebb639

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11435 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11767 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12876 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10705 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11246 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13814 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11420 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13620 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11386 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12271 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/9487 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13284 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9565 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10163 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17363 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10635 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10318 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13539 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10746 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8833 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9916 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2683 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14190 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10599 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderElement.cpp:
```diff
@@ -1922,7 +1922,7 @@ bool RenderElement::checkForRepaintDuringLayout() const
     if (document().view()->layoutContext().needsFullRepaint() || !everHadLayout() || hasSelfPaintingLayer())
         return false;
-    return !settings().repaintOutsideLayoutEnabled();
+    return true;
```
or you could just 
return everHadLayout()  && !document().view()->layoutContext().needsFullRepaint() && !hasSelfPaintingLayer()

but either way is fine.

anttijk committed:
Remove RepaintOutsideLayoutEnabled setting
https://bugs.webkit.org/show_bug.cgi?id=258570
rdar://111390032

Reviewed by Alan Baradlay.

Remove a long-dead experiment.

* Source/WebCore/page/Settings.yaml:
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::checkForRepaintDuringLayout const):

Canonical link: https://commits.webkit.org/265554@main
